torch/distributed/_tensor/ops/math_ops.py

```diff
@@ -350,6 +351,164 @@ def layer_norm_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> OpStrategy:
+        op_args_target_specs.append(mean_target_spec)
+        redistribute_costs.append(
+            generate_redistribute_costs(mean_strategy, mean_target_spec)
+        )
+        rstd_target_spec = rstd_strategy.strategies[idx].output_spec
+        op_args_target_specs.append(rstd_target_spec)
+        redistribute_costs.append(
+            generate_redistribute_costs(rstd_strategy, rstd_target_spec)
+        )
+
+        # arg: weight
+        # d_weight = sum(grad_out * (input - mean) / rstd, outer_dim, keepdim=False)
+        if output_mask[1]:
+            assert isinstance(weight_strategy, OpStrategy)
+            weight_src_spec = weight_strategy.strategies[idx].output_spec
+            weight_target_spec = DTensorSpec(
+                mesh=mesh,
+                placements=_replicate_dims_start_at(weight_src_spec.placements),
+                tensor_meta=weight_src_spec.tensor_meta,
+            )
+            op_args_target_specs.append(weight_target_spec)
+            redistribute_costs.append(
+                generate_redistribute_costs(weight_strategy, weight_target_spec)
+            )
+            d_weight_strategy.strategies.append(
+                PlacementStrategy(
+                    output_spec=weight_target_spec,
+                    input_specs=[],
+                    redistribute_cost=redistribute_costs,
+                )
+            )
+
+        # arg: bias
+        # d_bias = sum(grad_out, outer_dim, keepdim=False)
+        if output_mask[2]:
+            assert isinstance(bias_strategy, OpStrategy)
+            bias_src_spec = bias_strategy.strategies[idx].output_spec
+            bias_target_spec = DTensorSpec(
+                mesh=mesh,
+                placements=_replicate_dims_start_at(bias_src_spec.placements),
+                tensor_meta=bias_src_spec.tensor_meta,
+            )
+            op_args_target_specs.append(bias_target_spec)
+            redistribute_costs.append(
+                generate_redistribute_costs(bias_strategy, bias_target_spec)
+            )
+            d_bias_strategy.strategies.append(
+                PlacementStrategy(
+                    output_spec=bias_target_spec,
+                    input_specs=[],
+                    redistribute_cost=redistribute_costs,
+                )
+            )
+
+        if output_mask[0]:
+            # d_input is basically a product of element-wise mul of
+            # grad_out, rstd, and normalized input, among which rstd
+            # and normalized input (x_hat) should have the same sharding
+            # placements, and grad_out's sharding is determined by the
+            # pointwise result of x_hat and weight/bias.
+            d_input_strategy.strategies[-1].input_specs = op_args_target_specs
+
+        if output_mask[1]:
+            d_weight_strategy.strategies[-1].input_specs = op_args_target_specs
+
+        if output_mask[2]:
+            d_bias_strategy.strategies[-1].input_specs = op_args_target_specs
```

Comment:
hmmm thinking about this more, looks like the input specs are exactly the same for each OpStrategy output?

This make me think that TupleStrategy might be a overkill and headable to maintain for cases where we have "multiple different outputs that have different sharding strategies"

Maybe we should piggyback to OpStrategy, and simply make PlacementStrategy.output_spec be a `Union[DTensorSpec, Tuple[Optional[DTensorSpec], ...]]`